[PATCH] JSON::RPC::Legacy::Client: Do not try to parse non-existing JSON

gregoa opened the pull request:
In Debian we are currently applying the following patch to JSON-RPC.
We thought you might be interested in it too.

    From fc84cb2df7ef9cbc33f6fc03975b3a96f34bb9ff Mon Sep 17 00:00:00 2001
    From: Guillem Jover <guillem@hadrons.org>
    Date: Sun, 21 Jul 2024 23:27:16 +0200
    Subject: [PATCH] JSON::RPC::Legacy::Client: Do not try to parse non-existing
     JSON
    
    If the server reported a server error, then we get no JSON. We should
    return immediately instead of trying to parse the non-existing JSON,
    otherwise the callers will get confusing and truncated errors.

The patch is tracked in our Git repository at
https://salsa.debian.org/perl-team/modules/packages/libjson-rpc-perl/raw/master/debian/patches/0001-JSON-RPC-Legacy-Client-Do-not-try-to-parse-non-exist.patch

Thanks for considering,
  gregor herrmann,
  Debian Perl Group

guillemj committed:
JSON::RPC::Legacy::Client: Do not try to parse non-existing JSON

guillemj commented:
Hey @gregoa, thanks for forwarding the patch! (Sorry should have done that myself, and noticed now when I tried to do exactly that :). I think though the attribution in the patch in the MR is wrong? Also the [PATCH] prefix also seems to have slipped in.

gregoa commented:
On Mon, 17 Mar 2025 16:25:53 -0700, Guillem Jover wrote:

>Hey @gregoa, thanks for forwarding the patch! (Sorry should have 
>done that myself, and noticed now when I tried to do exactly that 
>:).

You're welcome.

> I think though the attribution in the patch in the MR is wrong? 
>Also the [PATCH] prefix also seems to have slipped in.

Hm, yeah, these seem to be unfortunate side effects of my workflow 
and our tooling … What seems to happen here is:

* I took your diff (which you probably produced with `git 
   format-patch`) from the Debian BTS and added it to the packing with 
   `quilt import`.
* I committed this file (and `d/p/series`) with you as the author:
   https://salsa.debian.org/perl-team/modules/packages/libjson-rpc-perl/-/commit/c07f7b9dff87f6cd656e43d36575cd5b8fbe0899
* Then I ran `dpt forward debian/patches/0001-JSON-RPC-Legacy-Client-Do-not-try-to-parse-non-exist.patch`
   (from `pkg-perl-tools`) which forks the upstream repo, transforms
   the patch (i.e. the diff file in `d/patches`) to a commit, pushes 
   it and creates a merge request.
* And here it adds the "[PATCH]", which makes sens for pure mails to 
   the CPAN RT but not so much; and your authorship is gone as the 
   commit in the forked repo, based on the patch file, is done by my 
   user …:
   https://github.com/pkg-perl-tools/JSON-RPC/commit/e6298e5ed46f0490169d4a7a594c4c62a9413db9
   whereas the info in the merge request at https://github.com/lestrrat-p5/JSON-RPC/pull/15
   still has the right information (again, this is taken from the diff 
   file.

So the combination of "the original patch is not some random diff but 
a nice git commit" with `quilt import` and then creating a new commit 
in a forked upstream repo with `dpt-forward` might have some room for 
improvement :)

https://salsa.debian.org/perl-team/modules/packages/pkg-perl-tools/-/blob/master/scripts/forward?ref_type=heads


I've now updated and force-pushed the commit in the forked upstream 
repo:
https://github.com/pkg-perl-tools/JSON-RPC/commit/bcf6d918b9cdb527923ec5b1646f78b4df955b90


Cheers,
gregor
   

-- 
  .''`.  https://info.comodo.priv.at -- Debian Developer https://www.debian.org
  : :' : OpenPGP fingerprint D1E1 316E 93A7 60A8 104D  85FA BB3A 6801 8649 AA06
  `. `'  Member VIBE!AT & SPI Inc. -- Supporter Free Software Foundation Europe
    `-

guillemj commented:
Thanks for the updated commit @gregoa ! :D
